test/pluck_test.rb

```diff
@@ -52,4 +53,26 @@ def test_symbol_options
     end
     assert_equal "Not available since :key depends on record", error.message
   end
+
+  def test_migrating
+    Robot.create!(name: "Test 1", email: "test1@example.org")
+    Robot.create!(name: "Test 2", email: "test2@example.org")
+
+    # migrated
+    assert_equal ["Test 1", "Test 2"], Robot.order(:id).pluck(:name)
+    assert_equal ["Test 1", "Test 2"], Robot.order(:id).pluck(:id, :name).map(&:last)
+    assert_equal [["Test 1", "test1@example.org"], ["Test 2", "test2@example.org"]], Robot.order(:id).pluck(:name, :email)
+
+    # half migrated
+    Robot.update_all(name_ciphertext: nil)
+    assert_equal ["Test 1", "Test 2"], Robot.order(:id).pluck(:name)
```

Comment:
Think we only need this assertion, as they'll all follow the same code path.

Reply:
I'm sorry, I fear I don't fully understand what you are referring to 🙈.

Do you mean condensing the whole test scenario (the whole `test_migrated` method) down to these two lines? Or would this be about condensing the `half migrated` block down to just one assertion?

You are right, they follow the same code path as they are essentially all covered by the guard clause in [this line](https://github.com/ankane/lockbox/pull/80/files#diff-61226409c83484faa78bff26dbb9541cd92332f0190a937e2aff4e43a5e88f79R8):
```ruby
return super unless lockbox_columns.any?
```

But wouldn't condensing down the whole test to just this single assertion mean the test relies on implicit knowledge about the internals of the `pluck` implementation?

I'm happy to update the test though, I just don't fully understand which change you have in mind ✌️

Reply:
I think we can have a strong test case with only one assertion (total). The behavior that's currently failing is when `migrating: true` is set and the ciphertext column is `nil`, so let's just test that.

Reply:
Alright 👍 

Thanks for the explanation 🙏. I reduced the test down to that assertion.